plugins/bitbucket-cloud-common/src/BitbucketCloudClient.ts

```diff
@@ -69,6 +69,67 @@ export class BitbucketCloudClient {
     );
   }
 
+  async createRepository(
+    workspace: string,
+    repo_slug: string,
+    repository: Models.Repository,
+  ): Promise<Models.Repository> {
+    const workspaceEnc = encodeURIComponent(workspace);
+    const url = this.createUrl(`/repositories/${workspaceEnc}/${repo_slug}`);
+
+    return await this.post(url, {
+      body: JSON.stringify(repository),
+    } as RequestInit).then((response: Response) =>
+      response
+        .json()
+        .then((repositoryResponse: Models.Repository) => repositoryResponse),
+    );
+  }
+
+  async updatePipelinesConfig(
+    workspace: string,
+    repo_slug: string,
+    pipelinesConfig: Models.PipelinesConfig,
+  ): Promise<Models.PipelinesConfig> {
```

Comment:
@pjungermann 
Do you have an opinion on "bitbucket-cloud common" returning a JSON object as a part of the OpenAPI spec?
I'm thinking to keep the common library generic, the data structure should be handled elsewhere, and only return a promise/request here. Ill leave another comment where i think it should go.